### PR TITLE
[docs][ci] added docs about GPU support out of the box for Windows wheels and small refactoring for dual test

### PR DESCRIPTION
**Pull request by StrikerRUS**
Continuation of #3403.

@tpboudreau  Please help with review.

### CODE REVIEW

File: docs/GPU-Windows.rst [hunk 1]
```diff
@@ -77,9 +77,9 @@ OpenCL SDK Installation
 Installing the appropriate OpenCL SDK requires you to download the correct vendor source SDK.
 You need to know what you are going to use LightGBM!
 
--  For running on Intel, get `Intel SDK for OpenCL`_ (NOT RECOMMENDED).
+-  For running on Intel, get `Intel SDK for OpenCL`_ (**NOT RECOMMENDED**).
```
Comment (StrikerRUS) [inline docs/GPU-Windows.rst:80]: @tpboudreau I think this is the reason why Intel OpenCL Runtime behaves oddly at Azure.

> I've run against the python package using three different versions the Intel CPU OpenCL platform. They all run successfully and give reasonable results, but for some reason they either sporadically hang at shutdown or terminate with the user script returning 'False'.

> **Warning**: using Intel OpenCL is not recommended and may crash your machine due to being non compliant to OpenCL standards.

File: docs/GPU-Windows.rst [hunk 2]
```diff
@@ -77,9 +77,9 @@ OpenCL SDK Installation
...
 
--  For running on AMD, get `AMD APP SDK`_ (you may want to replace the ``OpenCL.dll`` from GPU driver package with the one from the SDK, if the one shipped with the driver lacks some functions).
+-  For running on AMD, get AMD APP SDK (downloads `for Linux`_ and `for Windows`_) (you may want to replace the ``OpenCL.dll`` from GPU driver package with the one from the SDK, if the one shipped with the driver lacks some functions).
```
Comment (tpboudreau) [inline docs/GPU-Windows.rst:82]:
```suggestion
-  For running on AMD, get AMD APP SDK (downloads `for Linux`_ and `for Windows`_)   You may want to replace the ``OpenCL.dll`` from the GPU driver package with the one from the SDK, if the one shipped with the driver lacks some functions.
```

File: python-package/README.rst [hunk 3]
```diff
@@ -23,6 +23,8 @@ Install from `PyPI <https://pypi.org/project/lightgbm>`_ Using ``pip``
 
 You may need to install `wheel <https://pythonwheels.com>`_ via ``pip install wheel`` first.
 
+Compiled library that is included in the wheel file supports both **GPU** and **CPU** versions out of the box. This feature is experimental and available only for **Windows** currently. To use **GPU** version you only need to install OpenCL Runtime libraries. For NVIDIA and AMD GPU they are included in the ordinary drivers for your graphics card, so no action is required. For AMD or Intel CPU you can install `AMD APP SDK <https://github.com/microsoft/LightGBM/releases/download/v2.0.12/AMD-APP-SDKInstaller-v3.0.130.135-GA-windows-F-x64.exe>`_.
```
Comment (tpboudreau) [inline python-package/README.rst:26]:
```suggestion
Compiled library that is included in the wheel file supports both **GPU** and **CPU** versions out of the box. This feature is experimental and available only for **Windows** currently. To use **GPU** version you only need to install OpenCL Runtime libraries. For NVIDIA and AMD GPU they are included in the ordinary drivers for your graphics card, so no action is required. If you would like your AMD or Intel CPU to act like a GPU (for testing and debugging) you can install `AMD APP SDK <https://github.com/microsoft/LightGBM/releases/download/v2.0.12/AMD-APP-SDKInstaller-v3.0.130.135-GA-windows-F-x64.exe>`_.
```

**Comment by tpboudreau** [comment]
Overall, looks great.  Just two small suggestions.

And one quick question, just to clarify: will the published PyPI wheel now be built with the integrated-opencl flag, or is there some other change or PR required?

(I haven't forgotten that I'm going it to look into getting that build to default to device=gpu for the CI tests.  I'll get to that as soon as I can.)

File: docs/GPU-Windows.rst (hunk 2 above)
Comment (StrikerRUS) [reply docs/GPU-Windows.rst:82]: Added a dot.
```suggestion
-  For running on AMD, get AMD APP SDK (downloads `for Linux`_ and `for Windows`_). You may want to replace the ``OpenCL.dll`` from the GPU driver package with the one from the SDK, if the one shipped with the driver lacks some functions.
```

**Comment by StrikerRUS** [comment]
@tpboudreau 
> Overall, looks great. Just two small suggestions.

Thanks! Applied!

> will the published PyPI wheel now be built with the --integrated-opencl flag

Yeah! Actually, nightly releases are being built with this flag right now. You can check it.

image

https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html

Yesterday I run a few tests with them on my local machine with NVIDIA GPU and didn't notice any problems except the difference in returned results.

For ordinary releases we just take them for particular commit.
https://github.com/microsoft/LightGBM/blob/46986badbeb58b40f22e64b1ab72764cef1d553c/.vsts-ci.yml#L179-L195

> (I haven't forgotten that I'm going it to look into getting that build to default to device=gpu for the CI tests. I'll get to that as soon as I can.)

Great! Many thanks!

**Review by jameslamb — APPROVED** [review]
All looks great to me!

**Pull request merged**